src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -128,9 +128,13 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
 
                             var result = await _fhirService.UpdateResourceAsync(newObservation).ConfigureAwait(false);
 
-                            if (result.VersionId != "1")
+                            // If the version id is not equal to "1" on a create then it is likely two processes modified

```

Comment:
Yeah a pure create will first produce a version of 1

If two pure creates occur at the same time from different processes then the first version is "1" and the second version is "2" and wanted to detect the case where a version of "2" was created.

If the resource was created, and then soft-deleted, and then the connector creates the Observation again then the version returned will actually be "3". We do not want to log the metric in this case.

So I think we have to check that the version is equal to "2" in order to get the cases of two pure Creates occurring at the same time on a resource that has not been soft-deleted. It also means that we don't have a way to detect (create -> soft-delete -> then two creates at the same time) but I am ok with that only because that soft deleting in general should be rare and mainly used for testing purposes.

We previously had result.VersionId != "1" which is functionally the same as result.VersionId >= 2 but unfortunately that captures the soft-delete scenario (create -> soft-delete -> create) and the two simultaneous create scenario, but we only want to log the latter

Reply:
Oh I see. My first thought was `Couldn't we have X pure Creates at the same time`? For example if 3 pure creates are happening one of the processes will see Version == 3. And that wouldn't tick the metric.

I suppose in that case the process that hit Version == 2 would still tick the metric though. So at least we'd know that multiple processes were attempting the create. Just not how many.

Reply:
yes that is a good point - we could have X pure Creates at the same time. I am ok with not knowing exactly how many times or processes and just knowing whether the scenario is occurring or not. I think we've identified the exact part of the event hub sdk which is problematic for us so we should be able to address the issue, but should leave a metric here just in case so that we can confirm the scenario does not happen and that we have resolved it.